extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/config/TransitKeyConfig.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.vault.runtime.config;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class TransitKeyConfig {
+
+    /**
+     * Specifies the name of the key to use. By default this will be the property key alias. Used when
+     * the same transit key is used with different configurations. Such as:
+     * 
+     * <pre>
+     *     {@code
+     *     quarkus.vault.transit.key.foo1.name=foo
+     *     quarkus.vault.transit.key.foo1.key_version=1
+     *     quarkus.vault.transit.key.foo2.name=foo
+     *     quarkus.vault.transit.key.foo2.key_version=2
+     *     }
+     * </pre>
+     */
+    @ConfigItem
+    public Optional<String> name;
+
+    // sign
+
+    /**
+     * Set to true when the input is already hashed.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#prehashed">api documentation for prehashed</a>
+     */
+    @ConfigItem
+    public Optional<Boolean> prehashed;
+
+    /**
+     * When using a RSA key, specifies the RSA signature algorithm.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#signature_algorithm">api documentation for
+     *      signature_algorithm</a>
+     */
+    @ConfigItem
+    public Optional<String> signatureAlgorithm;
+
+    /**
+     * Specifies the way in which the signature should be marshaled.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#marshaling_algorithm">api documentation for
+     *      marshaling_algorithm</a>
+     */
+    @ConfigItem
+    public Optional<String> marshallingAlgorithm;
+
+    /**
+     * Specifies the hash algorithm to use for supporting key types.
+     * applies to sign operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#hash_algorithm">api documentation for
+     *      hash_algorithm</a>
+     */
+    @ConfigItem
+    public Optional<String> hashAlgorithm;
+
+    // encrypt
+
+    /**
+     * Specifies the type of key to create for the encrypt operation.
+     * applies to encrypt operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#type">api documentation for type</a>
+     */
+    @ConfigItem
+    public Optional<String> type;
+
+    /**
+     * If enabled, the key will support convergent encryption, where the same plaintext creates the same ciphertext.
+     * applies to encrypt operation.
+     * <p>
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#convergent_encryption">api documentation for
+     *      convergent_encryption</a>
+     */
```

Comment:
I've read the docs and I can't understand the point, perhaps we can drop it ? looks like it needs a `derived` property too, but not sure we need all of it

Reply:
there a whole [paragraph](https://www.vaultproject.io/docs/secrets/transit/index.html#convergent-encryption) in the documentation.

from the user's perspective the important pieces are:

> Convergent encryption is a mode where the same set of plaintext+context always result in the same ciphertext.
> ...
> This has many practical uses. A key usage mode is to allow values to be stored encrypted in a database, but with limited lookup/query support, so that rows with the same value for a specific field can be returned from a query.

by experience, it is always a pain to store encrypted data in db, and still allow some level of querying (eg search by lastname). 
the use case seems relevant to me.

the `derived` attribute is only relevant to the key creation. specifically when you create a key with convergent encryption, you need to do:
`vault write transit/keys/my-key convergent_encryption=true derived=true ...`
but this is really an admin thing. the vault extension (and the use) do not have to worry about that.

Reply:
@vsevel OK, thanks, hope it will be of help to some users